RoT/Protocol/RFCs/0003-RFC_PFM_Supported_Versions_Extension.md

```diff
@@ -0,0 +1,62 @@
+* Name: Extension for the PFM Supported Versions Command
+* Date: 2021-05-11
+* Pull Request: [#18](https://github.com/opencomputeproject/Security/pull/18)
+
+# Objective
+
+In the original design, there was only a single firmware component supported in
+a PFM.  The command to get the list of supported versions in a PFM didn't require
+any additional granularity.  With the introduction of version 2 of the PFM
+structure, there can now be multiple independent firmware components, each with
+their own list of supported firmware versions.  It is not always useful to get
+the full list of all versions for all components within a PFM.  Returning
+supported versions in this way would make it more challenging to search for a
+specific version of a specific component.  There needs to be a way to target a
+specific component's supported versions.
+
+# Proposal
+
+Update the Get Platform Firmware Manifest Supported Firmware command to have
+optional fields added to the request to specify the firmware component ID to
+query.  If these optional fields are not provided, every version for every
+component will be returned.
+
+# Specification Changelist
+
+The request structure for Get Platform Firmware Manifest Supported Firmware
+(https://github.com/opencomputeproject/Security/blob/master/RoT/Protocol/Challenge_Protocol.md#get-platform-firmware-manifest-supported-firmware)
+will be updated to include:
+1. A single byte length field that will specify the length of the firmware ID.
+2. A variable length, null-terminated firmware ID string.
```

Comment:
So just to be clear, would the new message look like this?
```
struct Req {
  u8 port;
  bool staging;
  u32 offset;
  u8 len;
  u8 firmware_id[len];
}
```

This is one of the commands I'm a bit less familiar with. I'm not sure what `offset` is for, nor what the fields of the response mean; can you elaborate on this for me in a response to this comment? (I just want to make sure my analysis of the change is sane.)

Reply:
That structure looks right, so long as bool is a 1-byte value.  The last two fields (len, firmware_id) are not required to be present.

The response is a list of null-terminated version strings pulled out of the PFM.  The first two fields in the response are the same as the PFM ID response.  Since a list of version strings can potentially be long, the offset is used to handle multi-message responses (just like reading the log).  The PFM ID is provided in the response so you know if the PFM you are querying has changed on you in the multi-message response case.